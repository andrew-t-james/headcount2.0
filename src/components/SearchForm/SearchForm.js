import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      query: ''
    };
  }

  handleChange = event => {
    const { value } = event.target;

    this.setState({ query: value });
  }

  handleSubmit = event => {
    event.preventDefault();
    const { query } = this.state;

    this.props.filterSchools(query);
    this.setState({ query: '' });
  }

  render() {
    const { query } = this.state;

    return (
      <form
        className="search-form"
        onSubmit={this.handleSubmit}>
        <input
          onChange={this.handleChange}
          type="text"
          name="query"
          value={query}
        />
        <button>Search</button>
      </form>
    );
  }
}

SearchForm.propTypes = {
  filterSchools: PropTypes.func.isRequired
};

export default SearchForm;
