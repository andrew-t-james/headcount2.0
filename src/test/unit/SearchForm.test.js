import React from 'react';
import { shallow, mount } from 'enzyme';
import SearchForm from '../../components/SearchForm/SearchForm';

describe('Search From unit test suite', () => {
  let wrapper;
  let filterSchoolsMock;
  beforeEach(() => {
    filterSchoolsMock = jest.fn();
    wrapper = shallow(<SearchForm filterSchools={filterSchoolsMock} />);
  });

  afterEach(() => wrapper.unmount());

  test('should initial state query of an empty string', () => {
    const mockState = {
      query: ''
    };

    expect(wrapper.state()).toEqual(mockState);
  });

  test('should update state on change', () => {
    const spy = spyOn(wrapper.instance(), 'handleChange');
    const mockEvent = {
      target: {
        value: 'Colorado'
      }
    };

    wrapper.find('input').simulate('change', mockEvent);
    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('query')).toBe('Colorado');
    expect(spy).toHaveBeenCalled();
  });

  test('should invoke filterSchools method and clear state on submit', () => {
    const spy = spyOn(wrapper.instance(), 'handleSubmit');
    const mockEvent = {
      preventDefault: jest.fn()
    };

    wrapper.find('form').simulate('submit', mockEvent);
    wrapper.instance().handleSubmit(mockEvent);

    expect(spy).toHaveBeenCalled();
    expect(wrapper.state('query')).toBe('');
    expect(filterSchoolsMock).toHaveBeenCalled();
  });

  test('matches sanpshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
