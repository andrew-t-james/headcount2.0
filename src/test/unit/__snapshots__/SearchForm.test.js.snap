// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search From unit test suite matches sanpshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm
    filterSchools={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="query"
          onChange={[Function]}
          type="text"
          value=""
/>,
        <button>
          Search
</button>,
      ],
      "className": "search-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "query",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search",
        },
        "ref": null,
        "rendered": "Search",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="query"
            onChange={[Function]}
            type="text"
            value=""
/>,
          <button>
            Search
</button>,
        ],
        "className": "search-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "query",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
